pkg/table-rule-selector/trie_selector.go

```diff
@@ -76,20 +84,111 @@ type trieSelector struct {
 }
 
 type node struct {
-	characters         map[byte]*item
-	asterisk, question *item
+	characters map[byte]item
+	asterisk   item
+	question   item
+	rItems     []item
 }
 
-type item struct {
-	child *node
+type item interface {
+	child() *node
+	setChild(*node)
+	getRule() interface{}
+	setRule(interface{})
+	getNextLevel() *node
+	setNextLevel(*node)
+}
+
+type baseItem struct {
+	ch *node
 
 	rule interface{}
 	// schema level ->(to) table level
 	nextLevel *node
 }
 
+func (i *baseItem) child() *node {
+	return i.ch
+}
+
+func (i *baseItem) setChild(c *node) {
+	i.ch = c
+}
+
+func (i *baseItem) getRule() interface{} {
+	return i.rule
+}
+
+func (i *baseItem) setRule(rule interface{}) {
+	i.rule = rule
+}
+
+func (i *baseItem) getNextLevel() *node {
+	return i.nextLevel
+}
+
+func (i *baseItem) setNextLevel(c *node) {
+	i.nextLevel = c
+}
+
 func newNode() *node {
-	return &node{characters: make(map[byte]*item)}
+	return &node{characters: make(map[byte]item)}
+}
+
+type ran struct {
+	start, end byte
+	hasBetween bool
+}
+
+type rangeItem struct {
+	*baseItem
+
+	hasNot bool
+	ranges []ran
+}
+
+func (i *rangeItem) match(i2 *rangeItem) bool {
```

Comment:
This function is only used to check whether `rangeItem`s are equivalent, by checking a covers b and b covers a, is it better to extract to a function like `equals`. Besides can we sort the `ran` in `rangeItem` and compare the `range` directly

Reply:
OK~

Reply:
I prefer the original `match` function and add an `equal` function to call the`match` function twice. Because here are some situations like `[a-c] == [a-ca-c]`.